src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ interface Item {
   [key: string]: any;
 }
 
-export function groupByKey(items, field) {
-  // write code here;
+export function groupByKey(items: Item[], field: keyof Item):Item {
+  const resultList: Item = {};
```

Comment:
The result should be not of an `Item` type. It should be an array of 'items'

Create the interface 
```
{
  [key: string] : Item[],
}
```

and use it as a result type